src/main/org/codehaus/groovy/runtime/DefaultGroovyMethods.java

```diff
@@ -5169,12 +5170,13 @@ public static Map groupBy(Object[] self, List<Closure> closures) {
      * @return a new Map grouped by keys
      * @since 1.0
      */
-    public static <G, K, V> Map<G, Map<K, V>> groupBy(Map<K, V> self, @ClosureParams(MapEntryOrKeyValue.class) Closure<G> closure) {
-        final Map<G, List<Map.Entry<K, V>>> initial = groupEntriesBy(self, closure);
-        final Map<G, Map<K, V>> answer = new LinkedHashMap<G, Map<K, V>>();
-        for (Map.Entry<G, List<Map.Entry<K, V>>> outer : initial.entrySet()) {
+    public static <G, K, V> Map<G, Map<? extends  K, ? extends V>> groupBy(
+            Map<K, V> self, @ClosureParams(MapEntryOrKeyValue.class) Closure<G> closure) {
```

Comment:
we should use the following guideline: no wildcards on self or returns, only on parameters we use as input. In mutating calls, even the (in/)out parameters should not have wildcards. Because wildcards make it very very difficult to use the container for any adding.